src/webnn_native/ops/Gemm.cpp

```diff
@@ -34,25 +34,38 @@ namespace webnn_native { namespace op {
         }
     }
 
+    MaybeError Gemm::CalculateShape() {
+        auto inputAShape = mInputs[0]->Shape();
+        auto inputBShape = mInputs[1]->Shape();
+        std::vector<int32_t> outputShape(2);
+        outputShape[0] = mOptions.aTranspose ? inputAShape[1] : inputAShape[0];
+        outputShape[1] = mOptions.bTranspose ? inputBShape[0] : inputBShape[1];
```

Comment:
I suppose it should check whether the two input shapes could be matrix multiplied. The spec says:

> The first input 2-D tensor with shape [M, K] if aTranspose is false, or [K, M] if aTranspose is true.

> The second input 2-D tensor with shape [K, N] if bTranspose is false, or [N, K] if bTranspose is true.

You should check whether K is same in the two input tensors.

Reply:
Yes, I will, thanks! @huningxin